config/accounts.go

```diff
@@ -1,9 +1,61 @@
 package config
 
+// RequestType : Request type enumeration
+type RequestType string
+
+// The request types
+const (
+	RequestTypeAMP   RequestType = "AMP"
+	RequestTypeApp   RequestType = "app"
+	RequestTypeVideo RequestType = "video"
+	RequestTypeWeb   RequestType = "web"
+)
+
 // Account represents a publisher account configuration
 type Account struct {
-	ID            string      `mapstructure:"id" json:"id"`
-	Disabled      bool        `mapstructure:"disabled" json:"disabled"`
-	CacheTTL      DefaultTTLs `mapstructure:"cache_ttl" json:"cache_ttl"`
-	EventsEnabled bool        `mapstructure:"events_enabled" json:"events_enabled"`
+	ID            string      `json:"id"`
+	Disabled      bool        `json:"disabled"`
+	CacheTTL      DefaultTTLs `json:"cache_ttl"`
+	EventsEnabled bool        `json:"events_enabled"`
+	GDPR          AccountGDPR `json:"gdpr"`
+}
+
+// AccountGDPR represents account-specific GDPR configuration
+type AccountGDPR struct {
+	Enabled            *bool                  `json:"enabled,omitempty"`
```

Comment:
I am curious why these fields are pointers, unlike the other "enabled" fields which are just bool

Reply:
Indeed. This is seems to be fairly standard Go convention to workaround the issue. I agree though, requiring use of a pointer to express "is present in JSON" is a fairly large code smell.